### PR TITLE
feat(rbac): display administration to authorized users

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://github.com/janus-idp/backstage-plugins/issues/842

**Test setup:**
```app-config.yaml```
```
auth:
  providers:
    github:
      development:
        clientId: $GITHUB_CLIENT_ID
        clientSecret: $GITHUB_CLIENT_SECRET
        
permission:
  enabled: true
  rbac:
    admin:
      users:
        - name: user:default/<your-github-id>
        - name: group:default/admins
```

```DynamicRoot.tsx```
```
Add SignInPage component 

    if (!app.current) {
      app.current = createApp({
        apis,
        components: {
          SignInPage: props => (
            <SignInPage
              {...props}
              auto
              provider={{
                id: 'github-auth-provider',
                title: 'GitHub',
                message: 'Sign in using GitHub',
                apiRef: githubAuthApiRef,
              }}
            />
          ),
        },
```

```auth.ts```
```
      github: providers.github.create({
        signIn: {
          resolver({ result }, ctx) {
            const userRef = `user:default/${result.fullProfile.username}`;
            return ctx.issueToken({
              claims: {
                sub: userRef, // The user's own identity
                ent: [userRef], // A list of identities that the user claims ownership through
              },
            });
          },
          // resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
        },
      }),
```

**_To load the Administration sidebar item dynamically_**

Test Setup

Add the `@janus-idp/backstage-plugin-rbac` package in the packages/app/package.json
Keep the app-config changes from the above set-up

app-config.yaml

```
dynamicPlugins:
  frontend:
    janus.dynamic-frontend-plugin:
      dynamicRoutes:
        …
        - path: '/rbac'
          importName: 'RbacPage'
          module: 'Administration'


```

Root.tsx

```
	export const Root = ({ children }: PropsWithChildren<{}>) => (
	…
        <ScalprumComponent
        scope="janus.dynamic-frontend-plugin"
        module="Administration"
        importName="Administration"
      />
	…
```

dynamic-modules-registry.json

```
      {
   	"module": "Administration",
   	 "scope": "janus.dynamic-frontend-plugin",
    	"manifestLocation": "http://localhost:8004/plugin-manifest.json"
  	}
```

dynamic-frontend-plugin/package.json

```
"scalprum": {
    …
    "exposedModules": {
       …
      "Administration": "./src/remotes/Administration"
    }
  }
```

Create Administration/index.ts under dynamic-frontend-plugin/src/remotes

`export * from '@janus-idp/backstage-plugin-rbac';`

**GIF:**

https://github.com/janus-idp/backstage-plugins/assets/22490998/69b8c23a-2f74-4db0-8839-222748f40f66

Updated the Administration Nav icon
Screenshot 2023-11-07 at 10 19 18 PM

### CODE REVIEW

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -66,6 +69,17 @@ export const AppBase = () => {
...
+              path="/rbac"
+              element={
+                <RequirePermission
+                  permission={policyEntityReadPermission}
+                  resourceRef="policy-entity"
+                >
+                  <RbacPage />
+                </RequirePermission>
+              }
+            />
```
Comment (invincibleJai) [inline packages/app/src/App.tsx:82]: good to add these in README and we can remove these changes in `**/app/**`

Comment (invincibleJai) [reply packages/app/src/App.tsx:82]: @tumido does the above comment make sense or this is an RBAC plugin so the change is fine?

**Comment by invincibleJai** [comment]
@debsmita1 and frontend dynamic plugin support is there now so pls ensure we test that flow as well and capture it as part of the PR description

**Comment by ShiranHi** [comment]
@debsmita1 looks good to me, thank you!

File: packages/app/src/App.tsx (hunk 1 above)
Comment (divyanshiGupta) [reply packages/app/src/App.tsx:82]: @invincibleJai your previous comment makes sense as this change will not be released with RBAC plugin as its in the core App component and the user will anyways need to add this rbac <Route /> manually in their App component.

File: packages/app/src/components/Root/Root.tsx [hunk 2]
```diff
@@ -77,6 +79,7 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       </SidebarGroup>
       <SidebarSpace />
       <SidebarDivider />
+      <Administration />
```
Comment (divyanshiGupta) [inline packages/app/src/components/Root/Root.tsx:82]: https://github.com/janus-idp/backstage-plugins/pull/895/files#r1386635365 - Same applies for this

File: packages/app/package.json [hunk 3]
```diff
@@ -47,6 +47,8 @@
     "@backstage/plugin-techdocs-module-addons-contrib": "^1.1.1",
     "@backstage/plugin-techdocs-react": "^1.1.12",
     "@backstage/plugin-user-settings": "^0.7.11",
+    "@janus-idp/backstage-plugin-rbac": "1.1.2",
+    "@janus-idp/backstage-plugin-rbac-common": "1.0.1",
```
Comment (divyanshiGupta) [inline packages/app/package.json:51]: @debsmita1 We havent added other plugins here so do we need to add these?

**Comment by debsmita1** [comment]
@divyanshiGupta @invincibleJai I have removed the changes from the app folder. Added ReadMe. And documented the steps to load the Administration sidebar item dynamically in the ReadMe. PTAL

**Comment by debsmita1** [comment]
Some existing tests are failing, working on them

**Review by invincibleJai — APPROVED** [review]
LGTM

Thanks @debsmita1 , let's fix the test

**Comment by AndrienkoAleksandr** [comment]
Build failed. Corresponding fix is here https://github.com/janus-idp/backstage-plugins/pull/929

**Review by invincibleJai — APPROVED** [review]
LGTM

**Pull request merged**